mediator-coordination/src/plugin.rs

```diff
@@ -1,29 +1,84 @@
-use crate::{util, web};
-
 use axum::Router;
 use did_endpoint::{didgen, util::filesystem::StdFileSystem};
+use mongodb::{error::Error as MongoError, options::ClientOptions, Client, Database};
 use server_plugin::{Plugin, PluginError};
+use std::sync::Arc;
+
+use crate::{
+    repository::stateful::coord::{MongoConnectionRepository, MongoSecretsRepository},
```

Comment:
Can the crate compile in these conditions:
- the `stateful` feature only is enabled;
- the `stateless` feature only is enabled;
- both features are enabled.

You'll certainly need to add feature annotations at many places, including here.

Reply:
Thanks for the help

Reply:
There are still stateful features compiled even when the `stateful` feature flag is not on. Do you think we should postpone this task?

Reply:
yes it will be best to go forward